### PR TITLE
API changes to support PDF conversion

### PR DESCRIPTION
**Pull request by loneil**
# Description
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-403

API changes supporting PDF exporting. Altered `template` field in the request body to use the following 3 new fields (removed old "filename" field)
- contentFileType **required**
- outputFileType
- outputFileName

Altered validation to support these fields.
Altered API docs and readme.

## Types of changes

New feature (non-breaking change which adds functionality) 
Documentation (non-breaking change with enhancements to documentation) 
Breaking change (fix or feature that would cause existing functionality to change) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the OpenAPI 3.0 `v*.api-spec.yaml` documentation (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/tests/unit/components/utils.spec.js [hunk 1]
```diff
@@ -42,3 +42,39 @@ describe('getFileExtension', () => {
   });
 });
 
+
+describe('determinOutputReportName', () => {
```
Comment (parc-jason) [inline app/tests/unit/components/utils.spec.js]: determin**e**OutputReportName

File: app/src/components/docGen.js [hunk 2]
```diff
@@ -15,9 +15,7 @@ const docGen = {
   generateDocument: (body, response) => {
     let tmpFile;
     try {
-      const inboundFileExtension = utils.getFileExtension(body.template.filename);
-
-      tmpFile = tmp.fileSync({ postfix: `.${inboundFileExtension}` });
+      tmpFile = tmp.fileSync({ postfix: `.${body.template.contentFileType}` });
```
[the remark this answers is not in the document]
Comment (parc-jason) [reply app/src/components/docGen.js]: i'd just drop the leading period, i think it's a bit heavy-handed to reject when we can easily handle it behind the scenes.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-idcqvl-tools.pathfinder.gov.bc.ca/dashboard?id=common-document-generation-service-pr-10

File: app/src/components/utils.js [hunk 3]
```diff
@@ -13,6 +15,17 @@ const utils = {
...
+
+  /** For the DocGen component, determine what the outputted (response) filename should be based
+   * on the template object from the request body,
+   *  @param {template} obj The template field from the request
+   *  @returns {string} The output filename for the response
+   */
+  determinOutputReportName: template => {
+    const extension = template.outputFileType ? template.outputFileType : template.contentFileType;
+    const name = template.outputFileName ? template.outputFileName : uuidv4();
+    return `${name}.${extension}`;
```
[the remark this answers is not in the document]
Comment (parc-jason) [reply app/src/components/utils.js]: i agree. and i think we are anticipating a refactor of the docGen. generateDocument method itself?  I don't like passing the body and response so far into the logic.  maybe leave in comments and address later.  basically get this pr up and out, so we can touch up DGRSC.

[the remark this answers is not in the document]
Comment (jujaga) [reply app/src/components/utils.js]: Sounds good. 👍

**Comment by loneil** [comment]
@parc-jason @jujaga thanks for reviewing and making changes. Agree about the extension-normalization and that the filename utility function shouldn't be in Utils on second thought. Good points.

**Pull request merged**